### ISSUE TITLE
[4.0][Bug] Column search for related attribute 

### ISSUE THREAD

**Issue by axelzuzek** [Laravel-Backpack/CRUD#2167]
# Bug report

After upgrading from backpack 3.6 to 4.0 related attribute search throws an error. 

### What I did
  $this->crud->addColumns([
       [
                'label' => "Owner",
                'name' => 'owner.name',
                'searchLogic' => function ($query, $column, $searchTerm) {
                    $query->orWhereHas('owner', function ($q) use ($column, $searchTerm) {
                        $q->where('name', 'like', '%'.$searchTerm.'%');
                    });
                }
            ],...

### What happened
Search throws error message: 
"Missing column type when trying to apply search term"

### Backpack, Laravel, PHP, DB version

backpack: ^4.0.0
laravel/framework": "6.0.*",
PHP 7.2.11
Mysql 5.7
When I run ```php artisan backpack:version``` the output is:

**Comment by justinmoh** [comment]
I think the issue is actually about:

```
SQLSTATE[23000]: Integrity constraint violation: 1052 Column 'name' in where clause is ambiguous
```

I ran into issue in this situation:

```php
CRUD::addColumn(
    [
        'label' => 'Location',
        'name' => 'location.name',
        'type' => 'text',
        'searchLogic' => function ($query, $column, $searchTerm) {
            $query->orWhereHas('location', function ($q) use ($column, $searchTerm) {
                $q->where('locations.name', 'like', '%'.$searchTerm.'%');
            });
        },
        'orderable' => true,
        'orderLogic' => function ($query, $column, $columnDirection) {
            return $query->leftJoin('locations', 'locations.id', '=', 'yards.location_id')
                ->orderBy('locations.name', $columnDirection)  // SQL query as `locations.name` column
                ->select('yards.*');
        }
    ]
);

CRUD::addColumn(
    [
        'label' => 'Name',
        'name' => 'name', // SQL query as bare `name`. When change to `yards.name` here, it couldn't render.
        'type' => 'text',
    ]
);
```

So to speak, there needs to be a way to prepend the original entity's attribute with its table name, or other methods avoid ambiguous where clause issue.

**Comment by pxpm** [comment]
@axelzuzek thanks for reporting this but i am afraid i could not reproduce your issue.

@tabacitu marked as bug, could you reproduce this ?

I will let you guys know what I did to try reproducing this issue.

Fresh demo in monster controller added:

```
 [
                'label' => "category",
                'name' => 'category.name',
                'searchLogic' => function ($query, $column, $searchTerm) {
                $query->orWhereHas('category', function ($q) use ($column, $searchTerm) {
                $q->where('name', 'like', '%'.$searchTerm.'%');
                });
                }
            ],
```
Searched and no errors, but as it has some other columns in this crud that allow search and the names used from Lorem Ipsum are just very similar, to make sure my search is working I removed all other columns from this crud and let this one be the only column.

Search by some category name in the search input, receive expected results. 

I hope you guys can provide me with some more info.

Best,
Pedro

**Comment by axelzuzek** [comment]
The problem seems to be fixed with backpack/crud ^4.0.12 - thx!
